packages/contracts-bedrock/test/setup/Upgrade.s.sol

```diff
@@ -0,0 +1,114 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+// Testing
+import { stdJson } from "forge-std/StdJson.sol";
+
+// Scripts
+import { Deployer } from "scripts/deploy/Deployer.sol";
+
+// Libraries
+import { Constants } from "src/libraries/Constants.sol";
+import { GameTypes } from "src/dispute/lib/Types.sol";
+import { EIP1967Helper } from "test/mocks/EIP1967Helper.sol";
+
+// Interfaces
+import { IFaultDisputeGame } from "interfaces/dispute/IFaultDisputeGame.sol";
+import { IDisputeGame } from "interfaces/dispute/IDisputeGame.sol";
+import { IDisputeGameFactory } from "interfaces/dispute/IDisputeGameFactory.sol";
+import { IAddressManager } from "interfaces/legacy/IAddressManager.sol";
+
+/// @title Upgrade
+/// @notice This script is called by Setup.sol as a preparation step for the foundry test suite, and is run as an
+///         alternative to Deploy.s.sol, when `UPGRADE_TEST=true` is set in the env.
+///         Like Deploy.s.sol this script saves the system addresses to disk so that they can be read into memory later
+///         on, however rather than deploying new contracts from the local source code, it simply reads the addresses
+///         from the superchain-registry.
+///         Therefore this script can only be run against a fork of a production network which is listed in the
+///         superchain-registry.
+///         This contract must not have constructor logic because it is set into state using `etch`.
+contract Upgrade is Deployer {
+    using stdJson for string;
+
+    /// @notice Returns the base chain name to use for forking
+    /// @return The base chain name as a string
+    function upgradeBaseChain() internal view returns (string memory) {
+        return vm.envOr("UPGRADE_BASE_CHAIN", string("mainnet"));
+    }
+
+    /// @notice Returns the OP chain name to use for forking
+    /// @return The OP chain name as a string
+    function upgradeOpChain() internal view returns (string memory) {
+        return vm.envOr("UPGRADE_OP_CHAIN", string("op"));
```

Comment:
Do we have a tracking issue for a (perhaps scheduled) CI job that runs the upgrade tests for every base chain <> OP Chain pairing in the superchain registry, that we hold the keys for?